gradio/flagging.py

```diff
@@ -344,3 +346,170 @@ def flag(
         self.repo.push_to_hub(commit_message="Flagged sample #{}".format(line_count))
 
         return line_count
+
+
+class HuggingFaceDatasetJSONSaver(FlaggingCallback):
+    """
+    A FlaggingCallback that saves flagged data to a HuggingFace dataset in JSONL format.
```

Comment:
Maybe add a line explaining why this is useful (e.g. saving the metadata json is useful for images/audio/other formats + it can be loaded out of the box with `datasets.load_dataset()`)